### PR TITLE
Kyles branch

### PR DESCRIPTION
**Pull request by KyleLuoma**
Deck returns an illegal card if user attempts to deal card from empty deck. Deck.init() input validation.

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tramstheman) [inline .classpath]: you may need to remove this from your git cache, but this file should be git ignored and not stored on the repo

File: .project [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tramstheman) [inline .project]: this file also should not belong on the repo

File: bin/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+/com/
```
Comment (tramstheman) [inline bin/.gitignore]: make sure the .classpath, .project, .idea, *.iml, etc.. are all ignored

File: src/com/northvine/engineering/cards/Deck.java [hunk 4]
```diff
@@ -101,16 +110,24 @@ public void shuffle()
...
     */
    public Card dealCard()
    {
-      Card pulledCard = new Card(inspectCard(topCard)); //stores a temp location of top card deep copy     
-      cards[topCard] = null; //sets the index of the cards array to null
-      topCard--; //shifts pointer to the index previous of where the chosen card was
-      return pulledCard;
+      if(topCard >= 0)
+      {
+         Card pulledCard = new Card(inspectCard(topCard)); //stores a temp location of top card deep copy
```
Comment (tramstheman) [inline src/com/northvine/engineering/cards/Deck.java]: this should be a final since its never re-instantiated

**Comment by KyleLuoma** [comment]
Made change to line 122 of Deck to make anonymous Card invocation final.
Removed unnecessary files

**Pull request merged**